api/src/test/java/org/pmiops/workbench/cohorts/CohortMaterializationServiceTest.java

```diff
@@ -0,0 +1,225 @@
+@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)
+@Transactional(propagation = Propagation.NOT_SUPPORTED)
+public class CohortMaterializationServiceTest {
+
+  @Autowired
+  private FieldSetQueryBuilder fieldSetQueryBuilder;
+
+  @Autowired
+  private AnnotationQueryBuilder annotationQueryBuilder;
+
+  @Autowired
+  private CdrBigQuerySchemaConfigService cdrBigQuerySchemaConfigService;
+
+  @Autowired
+  private ParticipantCohortStatusDao participantCohortStatusDao;
+
+  @Autowired
+  private CdrVersionDao cdrVersionDao;
+
+  @Autowired
+  private WorkspaceDao workspaceDao;
+
+  @Autowired
+  private CohortDao cohortDao;
+
+  @Autowired
+  private CohortReviewDao cohortReviewDao;
+
+  @Autowired
+  private ConceptDao conceptDao;
+
+  @Autowired
+  private ConceptSynonymDao conceptSynonymDao;
+
+  @PersistenceContext
+  private EntityManager entityManager;
+
+  private CohortReview cohortReview;
+  private CohortMaterializationService cohortMaterializationService;
+
+  @TestConfiguration
+  @Import({FieldSetQueryBuilder.class, AnnotationQueryBuilder.class,
+      TestBigQueryCdrSchemaConfig.class, CohortQueryBuilder.class,
+      CdrBigQuerySchemaConfigService.class, DomainLookupService.class})
+  @MockBean({BigQueryService.class})
+  static class Configuration {
+  }
+
+  @Before
+  public void setUp() {
+    CdrVersion cdrVersion = new CdrVersion();
+    cdrVersionDao.save(cdrVersion);
+
+    Workspace workspace = new Workspace();
+    workspace.setCdrVersion(cdrVersion);
+    workspace.setName("name");
+    workspace.setDataAccessLevelEnum(DataAccessLevel.PROTECTED);
+    workspaceDao.save(workspace);
+
+    Cohort cohort = new Cohort();
+    cohort.setWorkspaceId(workspace.getWorkspaceId());
+    cohort.setName("males");
+    cohort.setType("AOU");
+    Gson gson = new Gson();
+    cohort.setCriteria(gson.toJson(SearchRequests.males()));
+    cohortDao.save(cohort);
+
+    Cohort cohort2 = new Cohort();
+    cohort2.setWorkspaceId(workspace.getWorkspaceId());
+    cohort2.setName("all genders");
+    cohort2.setType("AOU");
+    cohort2.setCriteria(gson.toJson(SearchRequests.allGenders()));
+    cohortDao.save(cohort2);
+
+    cohortReview = new CohortReview();
+    cohortReview.setCdrVersionId(cdrVersion.getCdrVersionId());
+    cohortReview.setCohortId(cohort2.getCohortId());
+    cohortReview.setMatchedParticipantCount(3);
+    cohortReview.setReviewedCount(2);
+    cohortReview.setReviewSize(3);
+    cohortReviewDao.save(cohortReview);
+
+    participantCohortStatusDao.save(makeStatus(cohortReview.getCohortReviewId(), 1L, CohortStatus.INCLUDED));
+    participantCohortStatusDao.save(makeStatus(cohortReview.getCohortReviewId(), 2L, CohortStatus.EXCLUDED));
+
+    ConceptService conceptService = new ConceptService(entityManager, conceptDao, conceptSynonymDao);
+
+    this.cohortMaterializationService = new CohortMaterializationService(fieldSetQueryBuilder,
```

Comment:
Is there a reason you can't just `@Autowire` the materialization service rather than explicitly passing all its deps? `EntityManager` seems like the only potential stumbling block here, though I bet if you provided it in your test configuration...

Reply:
Will come back to this. The trouble is that ConceptService injects @PersistenceContext(unitName = "cdr")... i need to see how to set that up in test config.